### DISCUSSION TITLE
Regex for session domain does not support ip addresses

### DISCUSSION THREAD

**Discussion by RRebekka** [authelia/authelia#5737] [Q&A]
Hi, 

I'm currently testing the authelia v4.38.0-beta2 version, because I'm looking forward to the Multi-Domain Protection feature. I was wondering if it is also planned to support ip addresses as session domain values ( https://github.com/authelia/authelia/blob/0987da8524bd2e0c7cba5a00884beae67e0a6eb6/internal/configuration/validator/const.go#L426 ). I recognized that some styles of ip address formats are currently not supported. So for example, 127.0.0.1 is not allowed by the regex, but 127.0.0.10 is allowed. 

Would be great if the multi-domain protection would support that use case. 

Best regards

**Comment by james-d-elliott** [comment]
What is the actual use case behind this? Hosting multiple apps on 127.0.0.1? i.e. https://127.0.0.1/auth and https://127.0.0.1/app? Or something else?

**Reply by RRebekka** [reply]
Hi, 
thanks for your reply. I want to make authelia accessible via different ip addresses that are reverse proxied. So requests to e.g. 10.0.0.1/auth and 10.0.0.10/auth are proxied to 127.0.0.1:9091 where authelia is running. For that i have to set the session domain to 10.0.0.1 and another to 10.0.0.10. However, using for example 10.0.0.1 is not allowed due to the before mentioned regex rule. 

Best regards

**Reply by james-d-elliott** [reply]
I can aim to create a regex rule that handles that appropriately so that it can at least start. Is this something you've been previously using without issue? 

Also it should be noted that browsers will refuse to share session cookies between https://10.0.0.1 and any other IP, for example https://10.0.0.2, same applies for all IP addresses to my knowledge. For these reasons we'll only offer best-effort support, not direct support even if we allow it to be configured.

**Reply by RRebekka** [reply]
That would be great, if the regex rules allows to set that kind of ip addresses in the session domain. In the version 4.37.5 I was also able to use an ip address like e.g. 10.0.0.1 for the session domain. 

Yes, thanks for the hint about the refusion of session cookies between different IPs.
